ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx

```diff
@@ -30,7 +32,22 @@ const sortOptions = {
 function ListeningEndpointsPage() {
     const { page, perPage, setPage, setPerPage } = useURLPagination(10);
     const { sortOption, getSortParams } = useURLSort(sortOptions);
-    const { data, error, loading } = useDeploymentListeningEndpoints(sortOption, page, perPage);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const [searchValue, setSearchValue] = useState(() => {
+        const filter = searchFilter.Deployment;
+        return Array.isArray(filter) ? filter.join(',') : filter;
+    });
+
+    const { data, error, loading } = useDeploymentListeningEndpoints(
+        searchFilter,
```

Comment:
Hard to decide where to comment on unchanged code.

According to the pictures, pagination is still 1 - 10 after **central** matches only 2 deployments.

Reply:
Is there an edge case when data consists of less than the page size?
* when only one page of results
* for last page of results

Reply:
Yes this is an edge case caused by the indeterminate Pagination component variant due to the fact the we are not passing the total deployment count to the component.

At quick glance, it doesn't appear that there is a combination of props that will fix this issue. I'll make a note and investigate deeper, and if there isn't a good workaround we can revert to the typical pagination pattern we use.